### PR TITLE
[6.x] Core/Spell: fixed spell 114698 (Summon Amberleaf Troublemaker) and enabled SPELL_EFFECT_160

### PR DESCRIPTION
**Pull request by Lopfest**
**Changes proposed**:
- enabled SPELL_EFFECT_160 which should be forcecast
- fixes spell 114698, used in quest [29414](http://www.wowhead.com/quest=29414), which spawns NPCs in pattern below

pattern

**Target branch(es)**: 6x

**Tests performed**: successfully tested in-game

### CODE REVIEW

**Comment by MitchesD** [comment]
please rename SPELL_EFFECT_160 to that name as well ;) also one hint, how could to test it ingame? I didn't push all quest starters/enders in that zone

File: src/server/scripts/Pandaria/zone_the_wandering_isle.cpp [hunk 1]
```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2008-2016 TrinityCore <http://www.trinitycore.org/>
+ * Copyright (C) 2006-2009 ScriptDev2 <https://scriptdev2.svn.sourceforge.net/>
```
Comment (MitchesD) [inline src/server/scripts/Pandaria/zone_the_wandering_isle.cpp]: remove script dev2

**Comment by Lopfest** [comment]
Here is just a sample of sql for teseting.
```sql
UPDATE `creature` SET `PhaseId`=838 WHERE `Id`=59642;

UPDATE `creature_template` SET `AIName`='SmartAI' WHERE `entry`=59642;

DELETE FROM `phase_area` WHERE `AreaId`=5848;
INSERT INTO `phase_area` (`AreaId`, `PhaseId`, `Comment`) VALUES
(5848, 838, 'Pandaren starting zone - all classes');

DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`=13 AND `SourceEntry`=114710;
INSERT INTO `conditions` (`SourceTypeOrReferenceId`, `SourceGroup`, `SourceEntry`, `SourceId`, `ElseGroup`, `ConditionTypeOrReference`, `ConditionTarget`, `ConditionValue1`, `ConditionValue2`, `ConditionValue3`, `NegativeCondition`, `ErrorType`, `ErrorTextId`, `ScriptName`, `Comment`) VALUES
(13, 3, 114710, 0, 0, 31, 0, 4, 0, 0, 0, 0, 0, '', 'Forcecast Summon Amberleaf Troublemaker target Player');

DELETE FROM `smart_scripts` WHERE `entryorguid`=59642;
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
(59642, 0, 0, 0, 11, 0, 100, 0, 0, 0, 0, 0, 8, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Alysa Cloudsinger - On Respawn - Set React State'),
(59642, 0, 1, 0, 60, 0, 100, 0, 2000, 2000, 12000, 12000, 11, 114710, 0, 0, 0, 0, 0, 18, 40, 0, 0, 0, 0, 0, 0, 'Alysa Cloudsinger - Update - Cast Forcecast Summon Amberleaf Troublemaker');
```

I will post whole sql later when I finish at least first part of Wandeing Isle. But I'm a bit stucked because of #16758 :-(

**Comment by MitchesD** [comment]
I have scripted almost all quests in the first series, I am just stuck on some visual things

File: src/server/scripts/Pandaria/zone_the_wandering_isle.cpp [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                int32 radius;
+                if (urand(0, 2) == 0)
+                    radius = urand(0, 6);
+                else
+                    radius = 7;
+                float angle = M_PI * (urand(0, 7) / 7.f);
+                float x = 1181.75f + radius * sin(angle);
+                float y = 3444.5f + radius * cos(angle);
+
+                Position const SpawnPosition = { x, y, 102.9385f, 3.285759f };
```
Comment (joschiwald) [inline src/server/scripts/Pandaria/zone_the_wandering_isle.cpp]: could you use the previous destination target as reference point instead

Comment (Lopfest) [inline src/server/scripts/Pandaria/zone_the_wandering_isle.cpp]: Sorry, I didn't work with referenced variables yet. I have studied it a bit, but I still don't know where and why would you use it here. Could you send me desired line(s) with corrections please?

Comment (joschiwald) [inline src/server/scripts/Pandaria/zone_the_wandering_isle.cpp]: you should use GetHitDest() to get the spell target destination, then apply your position transformation

**Comment by Lopfest** [comment]
I really have no idea whats the purpose of the GetHitDest here and how I should use it, I hope its correct. If not, please advise.

**Comment by Lopfest** [comment]
Should I merge this branch with latest TC branch, or create two new PR separately (one for spell effect, one for spell script)? What else has to be done here? I need the pandaria script loader for another fixes.

**Comment by Treeston** [comment]
@Lopfest git rebase your branch on the latest mainline TC branch. That will reset the branch to latest mainline TC, then apply your commits from there.

https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase

**Comment by Shauren** [comment]
You should split the effect and script PRs, I have doubts about the correctness of it - the spell triggering it has target TARGET_UNIT_DEST_AREA_ENTRY, meaning it should have db conditions for proper selection

**Comment by Lopfest** [comment]
Ok i will close this this PR and create two separate.
About the spells: 114710 Forcecast Summon Amberleaf Troublemaker is always hitting players in range, and those players then cast the trigerred spell 114698 twice (two summons for each player)

**Comment by Shauren** [comment]
Yeah, that is why it needs conditions - are there invisible triggers spawned on proper destinations? Also it probably just needs the MaxAffectedTargets modification in SpellMgr

**Comment by Lopfest** [comment]
Yes for spell 114710 I already have condition to target only players.
On the spawn locations are no NPCs. About MaxAffectedTargets , I don't think its necessary. The more players are there, the more spawns will come. It is supposed to work that way.

**Comment by Ulduar** [comment]
For clarify:
Provide Spell Work WoD Data.
``` CPP
ID - 114710 Forcecast Summon Amberleaf Troublemaker
=================================================

Category = 0, SpellIconID = 1, activeIconID = 0, SpellVisual = (0,0)
Family SPELLFAMILY_GENERIC, flag [0] 0x00000000 [1] 0x00000000 [2] 0x00000000 [3] 0x00000000

SpellSchoolMask = 1 (SPELL_SCHOOL_MASK_NORMAL)
DamageClass = 0 (SPELL_DAMAGE_CLASS_NONE)
PreventionType = 0 (SPELL_PREVENTION_TYPE_NONE)
=================================================
Attributes: 0x29000180 (SPELL_ATTR0_HIDDEN_CLIENTSIDE, SPELL_ATTR0_HIDE_IN_COMBAT_LOG, SPELL_ATTR0_CASTABLE_WHILE_MOUNTED, SPELL_ATTR0_CASTABLE_WHILE_SITTING, SPELL_ATTR0_UNAFFECTED_BY_INVULNERABILITY)
AttributesEx1: 0x00000420 (SPELL_ATTR1_NOT_BREAK_STEALTH, SPELL_ATTR1_NO_THREAT)
AttributesEx2: 0x10484004 (SPELL_ATTR2_CAN_TARGET_NOT_IN_LOS, SPELL_ATTR2_UNK14, SPELL_ATTR2_NOT_NEED_SHAPESHIFT, SPELL_ATTR2_UNK22, SPELL_ATTR2_UNK28)
AttributesEx3: 0x00030300 (SPELL_ATTR3_ONLY_TARGET_PLAYERS, SPELL_ATTR3_TRIGGERED_CAN_TRIGGER_PROC_2, SPELL_ATTR3_CANT_TRIGGER_PROC, SPELL_ATTR3_NO_INITIAL_AGGRO)
AttributesEx4: 0x00800080 (SPELL_ATTR4_CAN_CAST_WHILE_CASTING, SPELL_ATTR4_UNK23)
AttributesEx5: 0x00060008 (SPELL_ATTR5_USABLE_WHILE_STUNNED, SPELL_ATTR5_USABLE_WHILE_FEARED, SPELL_ATTR5_USABLE_WHILE_CONFUSED)
AttributesEx6: 0x00001200 (SPELL_ATTR6_UNK9, SPELL_ATTR6_CASTABLE_WHILE_ON_VEHICLE)
=================================================
Spell Level = 0, base 0, max 0

Category = 0
DispelType = 0 (DISPEL_NONE)
Mechanic = 0 (MECHANIC_NONE)
SpellRange: (Id 1) "Self Only":
    MinRangeNegative = 0, MinRangePositive = 0
    MaxRangeNegative = 0, MaxRangePositive = 0

Cast time (Id 1) = 0,00

Interrupt Flags: 0x00000000, AuraIF 0x00000000, ChannelIF 0x00000000
Chance = 0, charges - 0
=================================================
Effect 0: Id 160 (SPELL_EFFECT_160) DIFFICULTY_NONE
BasePoints = 0
Targets (18, 8) (TARGET_DEST_CASTER, TARGET_UNIT_DEST_AREA_ENTRY)
Max Radius (Id 23) 40,00
   Trigger spell (114698) Призыв смутьяна из Янтарного Листа. Chance = 0

Effect 1: Id 160 (SPELL_EFFECT_160) DIFFICULTY_NONE
BasePoints = 0
Targets (18, 8) (TARGET_DEST_CASTER, TARGET_UNIT_DEST_AREA_ENTRY)
Max Radius (Id 23) 40,00
   Trigger spell (114698) Призыв смутьяна из Янтарного Листа. Chance = 0

ID - 114698 Призыв смутьяна из Янтарного Листа
=================================================

Triggred by spell: (114710) Forcecast Summon Amberleaf Troublemaker
Triggred by spell: (114710) Forcecast Summon Amberleaf Troublemaker
=================================================
Category = 0, SpellIconID = 1, activeIconID = 0, SpellVisual = (0,0)
Family SPELLFAMILY_GENERIC, flag [0] 0x00000000 [1] 0x00000000 [2] 0x00000000 [3] 0x00000000

SpellSchoolMask = 1 (SPELL_SCHOOL_MASK_NORMAL)
DamageClass = 0 (SPELL_DAMAGE_CLASS_NONE)
PreventionType = 0 (SPELL_PREVENTION_TYPE_NONE)
=================================================
Attributes: 0x29000100 (SPELL_ATTR0_HIDE_IN_COMBAT_LOG, SPELL_ATTR0_CASTABLE_WHILE_MOUNTED, SPELL_ATTR0_CASTABLE_WHILE_SITTING, SPELL_ATTR0_UNAFFECTED_BY_INVULNERABILITY)
AttributesEx1: 0x00000420 (SPELL_ATTR1_NOT_BREAK_STEALTH, SPELL_ATTR1_NO_THREAT)
AttributesEx2: 0x10484084 (SPELL_ATTR2_CAN_TARGET_NOT_IN_LOS, SPELL_ATTR2_UNK7, SPELL_ATTR2_UNK14, SPELL_ATTR2_NOT_NEED_SHAPESHIFT, SPELL_ATTR2_UNK22, SPELL_ATTR2_UNK28)
AttributesEx3: 0x10030200 (SPELL_ATTR3_TRIGGERED_CAN_TRIGGER_PROC_2, SPELL_ATTR3_CANT_TRIGGER_PROC, SPELL_ATTR3_NO_INITIAL_AGGRO, SPELL_ATTR3_UNK28)
AttributesEx4: 0x00800080 (SPELL_ATTR4_CAN_CAST_WHILE_CASTING, SPELL_ATTR4_UNK23)
AttributesEx5: 0x00060008 (SPELL_ATTR5_USABLE_WHILE_STUNNED, SPELL_ATTR5_USABLE_WHILE_FEARED, SPELL_ATTR5_USABLE_WHILE_CONFUSED)
AttributesEx6: 0x00001200 (SPELL_ATTR6_UNK9, SPELL_ATTR6_CASTABLE_WHILE_ON_VEHICLE)
AttributesEx8: 0x00000800 (SPELL_ATTR8_UNK11)
=================================================
Spell Level = 0, base 0, max 0

Category = 0
DispelType = 0 (DISPEL_NONE)
Mechanic = 0 (MECHANIC_NONE)
SpellRange: (Id 14) "Extra Long Range":
    MinRangeNegative = 0, MinRangePositive = 0
    MaxRangeNegative = 60, MaxRangePositive = 60

Cast time (Id 1) = 0,00
Duration: ID (28)  5000, 0, 5000
Interrupt Flags: 0x00000000, AuraIF 0x00000000, ChannelIF 0x00000000
Chance = 0, charges - 0
=================================================
Effect 0: Id 28 (SPELL_EFFECT_SUMMON) DIFFICULTY_NONE
BasePoints = 0
Targets (106, 0) (TARGET_DEST_CHANNEL_CASTER, NO_TARGET)
EffectMiscValueA = 59637
EffectMiscValueB = 2987
Radius (Id 48) 60,00
```

**Comment by Lopfest** [comment]
@Shauren you were right. MaxAffectedTargets was also required.

**Pull request closed without merging**